Content.Server/Clothing/Components/MagbootsComponent.cs

```diff
@@ -55,34 +53,14 @@ public void Toggle(EntityUid user)
             On = !On;
         }
 
-        void IUnequipped.Unequipped(UnequippedEventArgs eventArgs)
-        {
-            if (On && eventArgs.Slot == Slots.SHOES)
-            {
-                if (_entMan.TryGetComponent(eventArgs.User, out MovedByPressureComponent? movedByPressure))
-                {
-                    movedByPressure.Enabled = true;
-                }
-
-                if (_entMan.TryGetComponent(eventArgs.User, out ServerAlertsComponent? alerts))
-                {
-                    alerts.ClearAlert(AlertType.Magboots);
-                }
-            }
-        }
-
-        void IEquipped.Equipped(EquippedEventArgs eventArgs)
-        {
-            UpdateContainer();
-        }
-
-        private void UpdateContainer()
+        public void UpdateContainer()
         {
             if (!Owner.TryGetContainer(out var container))
                 return;
 
-            if (_entMan.TryGetComponent(container.Owner, out InventoryComponent? inventoryComponent)
-                && inventoryComponent.GetSlotItem(Slots.SHOES)?.Owner == Owner)
+            var invSystem = EntitySystem.Get<InventorySystem>();
+
+            if (invSystem.TryGetSlotEntity(container.Owner, "shoes", out var entityUid) && _entMan.GetComponent<TransformComponent>(entityUid.Value).ParentUid == Owner)
```

Comment:
I think something is wrong with this transform check here. If the magboots are equipped to the shoes slot, then `entityUid == Owner` so the transform check should always fail.

I think what it is supposed to be be doing is similar to what the `BreathToolComponent/LungSystem` did: check that the item is equipped in the correct slot in order to function (i.e., not just hands or pockets).

I imagine that will probably be a pretty common check that needs to be done, so it would probably also be nice if it was somehow managed consistently. But that might also take some effort and could be done separately, but this check should be fixed.

Reply:
ideally i'd like a way in the future that an item can just check which slots its in, for cases exactly like this